File: Add Code Here/Dart/bubblesort.dart
void bubbleSort(List<int> arr) {
  int n = arr.length;
  bool swapped;

  do {
    swapped = false;
    for (int i = 0; i < n - 1; i++) {
      if (arr[i] > arr[i + 1]) {
        // Swap arr[i] and arr[i + 1]
        int temp = arr[i];
        arr[i] = arr[i + 1];
        arr[i + 1] = temp;
        swapped = true;
      }
    }
  } while (swapped);
}

void main() {
  List<int> numbers = [64, 34, 25, 12, 22, 11, 90];

  print("Original list: $numbers");

  bubbleSort(numbers);

  print("Sorted list: $numbers");
}

File: Add Code Here/Python/bubble_sort.py
def bubble_sort(arr):
    n = len(arr)
    
    for i in range(n):
        # Flag to check if any swapping is done in this pass
        swapped = False
        
        # Last i elements are already in place, so we don't need to check them
        for j in range(0, n-i-1):
            if arr[j] > arr[j+1]:
                # Swap the elements
                arr[j], arr[j+1] = arr[j+1], arr[j]
                swapped = True
                
        # If no swapping occurred in this pass, the array is already sorted
        if not swapped:
            break

# Example usage
if __name__ == "__main__":
    arr = [64, 34, 25, 12, 22, 11, 90]
    
    print("Original array:", arr)
    bubble_sort(arr)
    print("Sorted array:", arr)
